### PR TITLE
ParserCommon tests

### PR DESCRIPTION
**Pull request by AntonHrbnts**
# Description

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

**Review by davsclaus — CHANGES_REQUESTED** [review]
Sorry but this test is complex to understand and would just cause pain for us to maintain in the future.

**Review by orpiske — CHANGES_REQUESTED** [review]
As @davsclaus said: the test code is a bit too complex. 

My suggestion: try to break the test in smaller pieces. For instance, separate the test setup part (i.e.: setting up mocks, expectations, etc) from the test result verification (asserting the result, logging when necessary, etc). 

Additionally, try to document to use JUnit's `@DisplayName` annotation to describe what the test does and what is the expected result (i.e.; `Tests that the code can consume null values and return an empty Box as expected`).

File: catalog/camel-route-parser/src/main/java/org/apache/camel/parser/helper/ParserCommon.java [hunk 1]
```diff
@@ -118,8 +118,8 @@ static FieldSource<JavaClassSource> getField(JavaClassSource clazz, Block block,
     public static boolean isNumericOperator(JavaClassSource clazz, Block block, Expression expression) {
         if (expression instanceof NumberLiteral) {
             return true;
-        } else if (expression instanceof SimpleName) {
-            FieldSource<JavaClassSource> field = getField(clazz, block, (SimpleName) expression);
+        } else if (expression instanceof SimpleName simpleName) {
+            FieldSource<JavaClassSource> field = getField(clazz, block, simpleName);
```
Comment (orpiske) [inline catalog/camel-route-parser/src/main/java/org/apache/camel/parser/helper/ParserCommon.java:122]: Also, this part here implies that there is something more other than tests. Maybe adjust the PR description, commit message and/or reference a Jira ticket if you are providing a fix.

File: catalog/camel-route-parser/src/test/java/org/apache/camel/parser/helper/ParserCommonTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            parserCommonMockedStatic.when(() -> ParserCommon.getField(clazz, block, expression)).thenReturn(fieldSource);
+            parserCommonMockedStatic.when(() -> ParserCommon.isNumericOperator(clazz, block, expression)).thenCallRealMethod();
+
+            assertFalse(ParserCommon.isNumericOperator(clazz, block, expression));
+
+            parserCommonMockedStatic.verify(() -> ParserCommon.getField(clazz, block, expression));
+            parserCommonMockedStatic.verify(() -> ParserCommon.isNumericOperator(clazz, block, expression));
+            parserCommonMockedStatic.verifyNoMoreInteractions();
+        }
+    }
```
Comment (orpiske) [inline catalog/camel-route-parser/src/test/java/org/apache/camel/parser/helper/ParserCommonTest.java:117]: I think I would probably move these 2 tests to a separate file. Then, I'd separate the test setup from the verification part and would use the `@DisplayName` annotation to make the expectations clearer.

**Comment by davsclaus** [comment]
closing inactive PR

**Pull request closed without merging**